packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -214,8 +214,21 @@ type ApplyFromHistoryAction = {
   query: BaseQuery;
 };
 
-export const applyFromHistory = (query: BaseQuery): ApplyFromHistoryAction => {
-  return { type: QueryBarActions.ApplyFromHistory, query };
+export const applyFromHistory = (
+  query: BaseQuery & { update?: Document }
+): QueryBarThunkAction<Promise<void>, ApplyFromHistoryAction> => {
+  return async (dispatch, getState, { localAppRegistry }) => {
+    dispatch({
+      type: QueryBarActions.ApplyFromHistory,
+      query,
+    } as ApplyFromHistoryAction);
```

Comment:
Why are we asserting here? The action shape already matches `ApplyFromHistoryAction` type